src/goto-symex/goto_symex_state.cpp

```diff
@@ -188,19 +188,22 @@ void goto_symex_statet::assignment(
   const auto level2_it =
     level2.current_names.emplace(l1_identifier, std::make_pair(lhs, 0)).first;
   symex_renaming_levelt::increase_counter(level2_it);
-  set_l2_indices(lhs, ns);
+  const renamedt<ssa_exprt, L2> l2_lhs = set_l2_indices(std::move(lhs), ns);
+  lhs = l2_lhs.get();
 
   // in case we happen to be multi-threaded, record the memory access
-  bool is_shared=l2_thread_write_encoding(lhs, ns);
+  bool is_shared = l2_thread_write_encoding(l2_lhs.get(), ns);
```

Comment:
It seems `lhs == l2_lhs.get()`, so why can't you just keep using `lhs` here (and below)?